tests/rptest/scale_tests/many_topics_test.py

```diff
@@ -1319,3 +1329,167 @@ def test_many_topics_config(self):
+        uri = f"{scheme}://{hostname}:8081/{path}"
+
+        if 'timeout' not in kwargs:
+            kwargs['timeout'] = 60
+
+        # Error codes that may appear during normal API operation, do not
+        # indicate an issue with the service
+        acceptable_errors = {409, 422, 404}
+
+        def accept_response(resp):
+            return 200 <= resp.status_code < 300 or resp.status_code in acceptable_errors
+
+        self.logger.debug(f"{verb} hostname={hostname} {path} {kwargs}")
+
+        # This is not a retry loop: you get *one* retry to handle issues
+        # during startup, after that a failure is a failure.
+        r = requests.request(verb, uri, **kwargs)
+        if not accept_response(r):
+            self.logger.info(
+                f"Retrying for error {r.status_code} on {verb} {path} ({r.text})"
+            )
+            time.sleep(10)
+            r = requests.request(verb, uri, **kwargs)
+            if accept_response(r):
+                self.logger.info(
+                    f"OK after retry {r.status_code} on {verb} {path} ({r.text})"
+                )
+            else:
+                self.logger.info(
+                    f"Error after retry {r.status_code} on {verb} {path} ({r.text})"
+                )
+
+        self.logger.info(
+            f"{r.status_code} {verb} hostname={hostname} {path} {kwargs}")
+
+        return r
+
+    def _post_subjects_subject_versions(self,
+                                        subject,
+                                        data,
+                                        headers=HTTP_POST_HEADERS,
+                                        **kwargs):
+        return self._request("POST",
+                             f"subjects/{subject}/versions",
+                             headers=headers,
+                             data=data,
+                             **kwargs)
+
+    def _get_subjects(self, deleted=False, headers=HTTP_GET_HEADERS, **kwargs):
+        return self._request("GET",
+                             f"subjects{'?deleted=true' if deleted else ''}",
+                             headers=headers,
+                             **kwargs)
+
+    def _get_schemas_ids_id_versions(self,
+                                     id,
+                                     headers=HTTP_GET_HEADERS,
+                                     **kwargs):
+        return self._request("GET",
+                             f"schemas/ids/{id}/versions",
+                             headers=headers,
+                             **kwargs)
+
+    def _get_schemas_ids_id_subjects(self,
+                                     id,
+                                     deleted=False,
+                                     headers=HTTP_GET_HEADERS,
+                                     **kwargs):
+        return self._request(
+            "GET",
+            f"schemas/ids/{id}/subjects{'?deleted=true' if deleted else ''}",
+            headers=headers,
+            **kwargs)
```

Comment:
Is any of this fundamentally different from `SchemaRegistryEndpoints`? Is it worth extracting that a bit so it can be shared?

Reply:
Looks reasonable at first glance, but I may defer to @michael-redpanda's intuition.